Library/ViewModels/DiscoveryViewModel.swift

```diff
@@ -194,7 +217,11 @@ public final class DiscoveryViewModel: DiscoveryViewModelType, DiscoveryViewMode
   }
 
   fileprivate let sortPagerSelectedSortProperty = MutableProperty<DiscoveryParams.Sort?>(nil)
+
+  // Stores last page before new page transition, default page is .magic, because it's the first page on Application Opened
+  fileprivate let previouslySortPagerSelectedSortProperty = MutableProperty<DiscoveryParams.Sort>(.magic)
   public func sortPagerSelected(sort: DiscoveryParams.Sort) {
+    self.previouslySortPagerSelectedSortProperty.value = self.sortPagerSelectedSortProperty.value ?? .magic
```

Comment:
I've noticed within this project, a given view model input does not interact with multiple properties so I don't think this line belongs here. I think you're trying to use the value of `sortPageSelectedSortProperty` before a new value is assigned but I think there should be a cleaner way to do that.